lsp-ui-doc.el

```diff
@@ -488,26 +457,31 @@ FN is the function to call on click."
         (lsp-ui-doc--put-click (match-beginning 0) (match-end 0)
                                'browse-url-at-mouse)))))
 
-(defun lsp-ui-doc--render-buffer (string symbol)
-  "Set the buffer with STRING."
+(defvar lsp-ui-doc--render-string nil
+  "The string to render in the documentation popup.")
+(defvar lsp-ui-doc--render-symbol nil
```

Comment:
Why you need this two global variables? The original version already pass `symbol` and `string`, why need to change function signature?

Reply:
I needed to call `lsp-ui-doc--render-buffer` from the `xwidget-event` callback lambda to solve the problem where webkit frames would be empty the first time they popped up.
https://github.com/emacs-lsp/lsp-ui/blob/61651bac3d3f932b11d549f5a0d557549369aa17/lsp-ui-doc.el#L651-L658

I can't just pass the parameters down to it, because it's only called when the webkit widget is first created, and since it's a lamba, it'll just capture the first values it sees, and won't update on subsequent hovers.

Reply:
> I needed to call `lsp-ui-doc--render-buffer` from the `xwidget-event` callback lambda to solve the problem where webkit frames would be empty the first time they popped up

Interesting. I wonder how just call `lsp-ui-doc--render-buffer` will resolve that problem?
From what I can see, the global vars are updated in `lsp-ui-doc--display`, which in turn will also call `lsp-ui-doc--render-buffer`.

Reply:
@Sorixelle

Reply:
The thing is, `lsp-ui-doc--render-buffer` inserts the text directly into the webkit widget via a javascript call, so it needs to be called after both the webkit widget has been inserted, and after `lsp-ui-doc.html` has finished loading. That last one in particular isn't guaranteed by the time `lsp-ui-doc--display` has finished executing, but that `'load-changed` event is fired specifically when the page has finished loading, so it's the best thing for us to use, unless there's something else I'm missing.